### ISSUE TITLE
Add a UDF for data validation based on questionnaire definitions 

### ISSUE THREAD

**Issue by yatharthranjan** [RADAR-base/ksql-extras#10]
KSQL UDF-
-
- Get and cache questionnaire definitions.
- Validate the current incoming data based on the definition.
- Special checking of each branching logic.
- Output Success or Error along with key, time and questionnaire name.

the stream and action-
-
- Create stream (using ksql-cli) for any questionnaire topics that require validation. If using the general questionnaire_response only one stream should be required. 
- Output topic can be same for all the input questionnaire topics.
- The radar-backend can read this output topic as input and inform admins or study team if there is an issue.

Any other requirements @mpgxvii, @afolarin ?

**Comment by afolarin** [comment]
I think it would be great to provide some guarantees of questionnaire definition/interpretation validation. But I can see this is not necessarily straightforward. I guess this goes beyond unit/integration/ software tests, we can test each questionnaire widget or even all current questionnaire definitions.
Considerations of where things can go wrong with a questionnaire:
 - the questionnaire doesn't render as expected either because of the app code or the underlying questionnaire's definition broken
 - the questionnaire branching logic doesn't evaluate as expected. I put this as a special case of "definition is broken" because complex branching might be difficult to test exhaustively.

When should the validations be done?
 - should this be done at the start when the definitions are created? Or when the code that touches questionnaire rendering is modified (smoke test)?
 - should this be done periodically to see if an error hasn't crept in
 - or done continuously for each submission?

**Comment by mpgxvii** [comment]
> When should the validations be done?

Yes, @yatharthranjan and I discussed about adding this to Github actions right when the definitions are created from REDCap (or manually). However, there could be instances where the definitions are in repositories where there's no GA setup (or also in the future when we can input definitions from other places not just Github).

**Comment by mpgxvii** [comment]
This would be good for validating different data types as well, for example checking if speech data is in the correct base64 format, etc.

**Comment by afolarin** [comment]
Also, perhaps the branching logic evaluating code in the app could do some sanity checks too and make it clear where something unexpected happens.
